pkg/reconciler/revision/resources/constants.go

```diff
@@ -29,6 +29,8 @@ const (
 	// TODO(mattmoor): Make this private once we remove revision_test.go
 	IstioOutboundIPRangeAnnotation = "traffic.sidecar.istio.io/includeOutboundIPRanges"
 
+	istioIncludeInboundPortsAnnotation = "traffic.sidecar.istio.io/includeInboundPorts"
```

Comment:
I have mixed feelings about this.  If we start to take advantage of this, then how might linkerd do the same?  I don't want to end up with a half dozen annotations for every possible mesh.

Do you have a specific objective in mind that needs this exposed?

Reply:
We are already exposing it technically. This is mostly a "fix" because we intercept arguably too much traffic today.

I full-heartedly agree that this is technically not the right place to do this. We do however lack a mechnism for pluggability here.

Reply:
If we have a smoking gun for the reported issue I am open to check the change in.  Also agree that we need a better mechanism to expose things like this rather, but this isn't much worse that before (we already have a bunch of other annotations).  May be we should file another issue for a cleaner interface for deployment annotations?

However I am still puzzled by the report - I believe the report was based on secondary signal (metrics was double-counted), which may or may not be caused by a port interception issue.